target/linux/mpc85xx/patches-4.14/104-powerpc-mpc85xx-change-P2020RDB-dts-file-for-OpenWRT.patch

```diff
@@ -0,0 +1,128 @@
+From 3465eb7c461a2d3cf4326352a139db43fbcc15fd Mon Sep 17 00:00:00 2001
+From: Pawel Dembicki <paweldembicki@gmail.com>
+Date: Sun, 30 Dec 2018 23:24:41 +0100
+Subject: [PATCH] powerpc: mpc85xx: change P2020RDB dts file for OpenWRT
+
+This patch apply chages for OpenWRT in P2020RDB
+dts file.
+
+Signed-off-by: Pawel Dembicki <paweldembicki@gmail.com>
+---
+ arch/powerpc/boot/dts/fsl/p2020rdb.dts | 43 +++++++++++++++++++++-----
+ 1 file changed, 35 insertions(+), 8 deletions(-)
+
+diff --git a/arch/powerpc/boot/dts/fsl/p2020rdb.dts b/arch/powerpc/boot/dts/fsl/p2020rdb.dts
+index 435a319958cb..41320958ce98 100644
+--- a/arch/powerpc/boot/dts/fsl/p2020rdb.dts
++++ b/arch/powerpc/boot/dts/fsl/p2020rdb.dts
+@@ -2,6 +2,7 @@
+  * P2020 RDB Device Tree Source
+  *
+  * Copyright 2009-2012 Freescale Semiconductor Inc.
++ * Copyright 2018      Pawel Dembicki <paweldembicki@gmail.com>
+  *
+  * This program is free software; you can redistribute  it and/or modify it
+  * under  the terms of  the GNU General  Public License as published by the
+@@ -9,10 +10,15 @@
+  * option) any later version.
+  */
+ 
++/dts-v1/;
++
++#include <dt-bindings/gpio/gpio.h>
++#include <dt-bindings/input/input.h>
++
+ /include/ "p2020si-pre.dtsi"
+ 
+ / {
+-	model = "fsl,P2020RDB";
++	model = "Freescale P2020RDB";
+ 	compatible = "fsl,P2020RDB";
+ 
+ 	aliases {
+@@ -25,6 +31,17 @@
+ 		pci1 = &pci1;
+ 	};
+ 
++	gpio_export {
```

Comment:
Assuming it isn't your intention to have the gpio available from userspace, [gpio-hog](https://www.kernel.org/doc/Documentation/devicetree/bindings/gpio/gpio.txt) is the upstream way to set GPIO early in the boot process to a defined state.

Beside of that, I would like to prevent adding our gpio-export hack to more targets. It was rejected upstream a long time ago and a lot of drivers are able to pull gpios on their own now days.

Maybe kernel commit bafbdd527d56 ("phylib: Add device reset GPIO support") does the job for you. A word of warning, I tried to use the new dts property on my own and failed back in the days (maybe it's fixed already):

  The per PHY reset GPIO code isn't able to bring a PHY out of reset,  which is initially in reset.
  
  The get_phy_device() in of_mdiobus_register_phy() returns an error, since the phy id for a PHY in reset can't be read (is 0xfffffff). The code deasserting the optional (initial) is called via phy_device_register() which is never reached in such a situation.

I have this issue on my todo list since april this year, but never managed to check if it can be easily fixed.

Reply:
I add support for 4.19 kernel and use reset-gpio from upstream. 

So far everything work as expected.